src/rebalancer/bridges/EverclearBridgeV2.sol

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2025 Merge Layers Inc.
+//
+// This source code is licensed under the Business Source License 1.1
+// (the "License"); you may not use this file except in compliance with the
+// License. You may obtain a copy of the License at
+//
+//     https://github.com/malda-protocol/malda-lending/blob/main/LICENSE-BSL
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity =0.8.28;
+
+/*
+ _____ _____ __    ____  _____ 
+|     |  _  |  |  |    \|  _  |
+| | | |     |  |__|  |  |     |
+|_|_|_|__|__|_____|____/|__|__|   
+*/
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {SafeApprove} from "src/libraries/SafeApprove.sol";
+import {BytesLib} from "src/libraries/BytesLib.sol";
+
+import {IBridge} from "src/interfaces/IBridge.sol";
+
+import {BaseBridge} from "src/rebalancer/bridges/BaseBridge.sol";
+
+interface IFeeAdapterV2 {
+  struct FeeParams {
+    uint256 fee;
+    uint256 deadline;
+    bytes sig;
+  }
+
+  struct Intent {
+    bytes32 initiator;
+    bytes32 receiver;
+    bytes32 inputAsset;
+    bytes32 outputAsset;
+    uint32 origin;
+    uint64 nonce;
+    uint48 timestamp;
+    uint48 ttl;
+    uint256 amount;
+    uint256 amountOutMin;
+    uint32[] destinations;
+    bytes data;
+  }
```

Comment:
Let's use the correct interface for `Intent`struct - it's imported from [`IEverclearV2`](https://github.dev/everclearorg/monorepo/blob/dev/packages/contracts/src/contracts/hub/HubStorageV2.sol), not `IFeeAdapterV2`.

Also, I feel like both of these interfaces should be a part of `src/interfaces/external/everclear`, rather than inlined within the contract. Always have a single contract/interface per file.